apps/ledger-live-mobile/src/screens/WalletCentricSections/MarketPrice.tsx

```diff
@@ -22,12 +24,30 @@ const MarketPrice = ({ currency, selectedCoinData, counterCurrency }: Props) =>
   const { locale } = useSettings();
   const navigation = useNavigation();
 
+  const [range] = useTimeRange();
+
   const goToMarketPage = useCallback(() => {
     navigation.navigate(ScreenName.MarketDetail, {
       currencyId: currency.id,
     });
   }, [currency, navigation]);
 
+  const getPrice = (selectedCoinData: CurrencyData, range: PortfolioRange) => {
+    switch (range) {
+      case "all":
+      case "year":
+        return selectedCoinData.priceChangePercentage["1y"];
+      case "month":
+        return selectedCoinData.priceChangePercentage["30d"];
+      case "week":
+        return selectedCoinData.priceChangePercentage["7d"];
+      default:
+      case "day":
+        return selectedCoinData.priceChangePercentage["24h"];
```

Comment:
Could you use values from KeysPriceChange instead of hardcoded values, please?
Maybe we could have a mapping here instead of a switch